src/browser/provider/host.js

```diff
@@ -0,0 +1,103 @@
+import { assignInWith } from 'lodash';
+import { getViewportSize } from 'testcafe-browser-natives';
+
+
+export default class BrowserProviderHost {
+    constructor (providerObject) {
+        assignInWith(this, providerObject, (hostProp, providerProp, propName) => {
+            var overloadedMethodName = '$' + propName;
+
+            if (this[overloadedMethodName]) {
+                this[overloadedMethodName] = providerProp;
+                return hostProp;
+            }
+
+            return providerProp;
+        });
+
+        this.evaluators          = [];
+        this.initialized         = false;
```

Comment:
I guess we need to hide our vars using symbols to avoid accidental overwriting, like it's done in `ReporterPluginHost`